specs/vm/instruction_set.md

```diff
@@ -937,6 +939,7 @@ Panic if:
 | Operation   | ```$rA = MEM[$rB, $rD] == MEM[$rC, $rD];``` |
 | Syntax      | `meq $rA, $rB, $rC, $rD`                    |
 | Encoding    | `0x00 rA rB rC rD`                          |
+| Effects     | Storage read                                |
```

Comment:
Ah so instructions that access `MEM[..]` do not actual read/write storage. They only access VM memory so they technically have no permanent side effects.

I believe the only instructions that access storage are `srw`, `srwq`, `sww`, `swwq`, `bal`, and `scwq` (Just added a we don't support yet in the compiler)

Reply:
This is really useful! Thank you